StitchCore-iOS/StitchCore-iOS/Services/Internal/NamedServiceClientProvider.swift

```diff
@@ -1,19 +1,39 @@
 import StitchCore
 
+/**
+ * A protocol describing a class that can provide clients for a particular named Stitch service.
+ */
 public protocol NamedServiceClientProvider {
-    associatedtype T
+    /**
+     * The type that this `ServiceClientProvider` can provide.
+     */
+    associatedtype ClientType
 
-    func client(forService service: StitchService,
-                withClient client: StitchAppClientInfo) -> T
+    /**
+     * Returns a client of type `ClientType`, with the provided `StitchService` and `StitchAppClientInfo` objects.
+     */
+    func client(forService service: StitchService, withClient client: StitchAppClientInfo) -> ClientType
 }
 
+/**
+ * A generic wrapper for a `NamedServiceClientProvider`.
+ */
 public struct AnyNamedServiceClientProvider<T> {
```

Comment:
[q] Why is this different than how the auth suppliers work?

Reply:
Not really sure. I didn't write this originally, and I figured we would make usability adjustments here once we actually implement some concrete services. This stuff isn't being used anywhere yet anyway.